Library/Homebrew/formula_cellar_checks.rb

```diff
@@ -275,6 +275,47 @@ def check_python_symlinks(name, keg_only)
     "Python formulae that are keg-only should not create `pip3` and `wheel3` symlinks."
   end
 
+  INVALID_EMPTY_FILE_NAMES = %w[AUTHORS ChangeLog changes COPYING NEWS README TODO].freeze
+  INVALID_EMPTY_FILE_EXTENSIONS = %w[.html .md .rst .txt].freeze
+  INVALID_EMPTY_FILE_LOCATIONS = %w[bin libexec share/man share/doc].freeze
+  # Names and extensions to ignore, even if they are in a location within prefix given above
+  VALID_EMPTY_FILE_NAMES = %w[__init__.py REQUESTED].freeze
+  VALID_EMPTY_FILE_EXTENSIONS = %w[.stamp].freeze
+
+  def check_empty_files(prefix)
+    return unless prefix.directory?
+
+    empty_files = []
+    prefix.children.each do |f|
+      next if INVALID_EMPTY_FILE_NAMES.none?(File.basename(f)) &&
+              INVALID_EMPTY_FILE_EXTENSIONS.none?(File.extname(f))
+      next unless File.zero?(f)
+
+      empty_files << f
+    end
+
+    INVALID_EMPTY_FILE_LOCATIONS.each do |loc|
+      next unless (prefix/loc).directory?
+
+      # search all directories in INVALID_EMPTY_FILE_LOCATIONS and their subdirectories
+      Dir.glob(prefix/loc/"**/*").each do |f|
+        next if VALID_EMPTY_FILE_NAMES.any?(File.basename(f)) ||
+                VALID_EMPTY_FILE_EXTENSIONS.any?(File.extname(f))
```

Comment:
```suggestion
        next unless VALID_EMPTY_FILE_NAMES.include?(f.basename)
        next unless VALID_EMPTY_FILE_EXTENSIONS.include?(f.extname)
```

Reply:
This ended up having the opposite effect than intended: checking only files with both valid names and extensions rather than ignoring them. I used `next if` statements instead, and it seems to work, though:

```
# search all directories in INVALID_EMPTY_FILE_LOCATIONS and their subdirectories
      Pathname.glob(path/"**/*").each do |p|
        next if VALID_EMPTY_FILE_NAMES.include?(p.basename(".*").to_s.delete_suffix("/"))
        next if VALID_EMPTY_FILE_EXTENSIONS.include?(p.extname)
        next unless File.zero?(p)

        empty_files << p
      end
```

Reply:
Yeh, that seems fine, another misunderstanding on my end.